stetho/src/main/java/com/facebook/stetho/inspector/elements/UIDOMProvider.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.TextView;
+
+import com.facebook.stetho.inspector.protocol.module.DOM;
+
+public final class UIDOMProvider implements DOM.Provider {
+  private final NodeDescriptorMap mNodeDescriptorMap;
+  private final Application mApplication;
+  private Listener mListener;
+
+  public UIDOMProvider(Context context) {
+    mApplication = (Application)context.getApplicationContext();
+
+    mNodeDescriptorMap = new NodeDescriptorMap()
```

Comment:
This is generally an anti-pattern on Android.  Pre-initializing a mapping data structure adds to the heap and adds init overhead.  Strongly recommend either lazily initializing on first access or using the imperative pattern (that is, a function that translates Class/Object instances to `*NodeDescriptor`)